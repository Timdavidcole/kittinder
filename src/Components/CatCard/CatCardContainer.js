import React from "react";
import shuffle from "../../Models/shuffle";
import CatCard from "./CatCard";


class CatCardContainer extends React.Component {
  render() {
    return (
      <div className="catCardContainer" style={{ padding: "10px" }}>
        {shuffle(this.props.cats).map((cat, index) => {
          return <CatCard cat={cat} key={index} />;
        })}
      </div>
    );
  }
}

export default CatCardContainer;
