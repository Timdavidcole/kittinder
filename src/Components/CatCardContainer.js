import React from "react";
import agent from "../agent";

class CatCardContainer extends React.Component {
  constructor(props) {
    super(props);
    this.state = { image: [] };
  }

  componentDidMount() {
    agent.Images.get({ breed_id: this.props.breed.id }).then(data =>
      this.setState({image: data[0].url})
    );
  }

  render() {
    const breed = this.props.breed;
    console.log(breed.id);

    return <div className={"catCardContainer"}>{breed.name}<img src={this.state.image}></img></div>;
  }
}

export default CatCardContainer;
