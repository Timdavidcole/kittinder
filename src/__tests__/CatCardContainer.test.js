import CatCardContainer from "../Components/CatCard/CatCardContainer";
import React from "react";
import renderer from "react-test-renderer";
import "@testing-library/jest-dom";
import { mount, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });
jest.mock("../Models/shuffle");

const breeds = ["Cat1", "Cat2", "Cat3"];

test("checks CatCardContainer snapshot", () => {
  const catCardContainer = renderer.create(<CatCardContainer cats={breeds} />);

  let tree = catCardContainer.toJSON();
  expect(tree).toMatchSnapshot();
});

test("checks is creates multiple CatCards", () => {
  const catCardContainer = mount(<CatCardContainer cats={breeds} />);
  expect(catCardContainer.find('#catCard')).toHaveLength(3);
});
