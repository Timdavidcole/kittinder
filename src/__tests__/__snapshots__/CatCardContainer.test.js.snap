// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checks CatCardContainer snapshot 1`] = `
<div
  className="catCardContainer"
  style={
    Object {
      "padding": "10px",
    }
  }
>
  <center>
    <div
      id="catCard"
      style={
        Object {
          "borderRadius": "5px",
          "boxShadow": "0px 0px 15px 0px rgba(158,158,158,1)",
          "margin": "20px",
          "maxWidth": "400px",
          "padding": "5px",
        }
      }
    >
      <h2 />
      <div
        id="errorMessage"
        style={
          Object {
            "fontSize": "0.8rem",
            "padding": "5px ",
          }
        }
      >
        This cat doesn't like having it's picture shown right now, please try again later.
      </div>
      <div
        style={
          Object {
            "width": "100%",
          }
        }
      >
        <span />
      </div>
    </div>
  </center>
  <center>
    <div
      id="catCard"
      style={
        Object {
          "borderRadius": "5px",
          "boxShadow": "0px 0px 15px 0px rgba(158,158,158,1)",
          "margin": "20px",
          "maxWidth": "400px",
          "padding": "5px",
        }
      }
    >
      <h2 />
      <div
        id="errorMessage"
        style={
          Object {
            "fontSize": "0.8rem",
            "padding": "5px ",
          }
        }
      >
        This cat doesn't like having it's picture shown right now, please try again later.
      </div>
      <div
        style={
          Object {
            "width": "100%",
          }
        }
      >
        <span />
      </div>
    </div>
  </center>
  <center>
    <div
      id="catCard"
      style={
        Object {
          "borderRadius": "5px",
          "boxShadow": "0px 0px 15px 0px rgba(158,158,158,1)",
          "margin": "20px",
          "maxWidth": "400px",
          "padding": "5px",
        }
      }
    >
      <h2 />
      <div
        id="errorMessage"
        style={
          Object {
            "fontSize": "0.8rem",
            "padding": "5px ",
          }
        }
      >
        This cat doesn't like having it's picture shown right now, please try again later.
      </div>
      <div
        style={
          Object {
            "width": "100%",
          }
        }
      >
        <span />
      </div>
    </div>
  </center>
</div>
`;
